src/Stack.cpp

```diff
@@ -0,0 +1,192 @@
+	: m_stack(std::move(other.m_stack)) {}
+
+// ======================== methods ==============================
+
+/**
+ * pushing a value to the stack
+ * @param value the value to push
+ **/
+void Stack::push(int value)
+{
+	m_stack.push_back(value);
+}
+
+/**
+ * poping a value from the stack
+ **/
+void Stack::pop()
+{
+	m_stack.pop_back();
+}
+
+/**
+ * check if the stack is empty
+ * @return true if the stack is empty, false otherwise
+ **/
+bool Stack::isEmpty() const
+{
+	return m_stack.empty();
+}
+
+// ==================== operators overloading ====================
+
+/**
+ * operator= assigns a stack to another stack
+ * @param other the stack to assign
+ * @return the stack after the assignment
+ **/
+Stack& Stack::operator=(const Stack& other)
+{
+	m_stack = other.m_stack;
+	return *this;
+}
+
+/**
+ * operator= move a stack to another stack
+ * @param other the stack to move
+ * @return the stack after the assignment
+ **/
+Stack& Stack::operator=(Stack&& other) noexcept
+{
+	m_stack = std::move(other.m_stack);
+	return *this;
+}
+
+/**
+ * operator== checks if two stacks are equal
+ * @param other the stack to compare to
+ * @return true if the stacks are equal, false otherwise
+ **/
+bool Stack::operator==(const Stack& other) const
+{
+	return m_stack == other.m_stack;
+}
+
+/**
+ * operator!= checks if two stacks are not equal
+ * @param other the stack to compare to
+ * @return true if the stacks are not equal, false otherwise
+ **/
+bool Stack::operator!=(const Stack& other) const
+{
+	return m_stack != other.m_stack;
+}
+
+/**
+ * operator += adds a value to all the elements in the stack
+ * @param value the value to add 
+ * @return the stack after the addition
+ **/
+Stack& Stack::operator+=(int value)
+{
+	for (auto& element : m_stack) {
+		element += value;
+	}
+	return *this;
+}
+
+/**
+ * operator += substract a value from all the elements in the stack
+ * @param value the value to substract
+ * @return the stack after the substraction
+ **/
+Stack& Stack::operator-=(int value)
+{
+	*this += -value;
+	return *this;
+}
+
+/**
+ * operator *= multiplies all values in the stack by a given value
+ * @param value the value to multiply by
+ * @return the stack after the multiplication
+ **/
+Stack& Stack::operator*=(int value)
+{
+	for (auto& element : m_stack) {
+		element *= value;
+	}
+	return *this;
+}
+
+/**
+ * operator /= divides all values in the stack by a given value
+ * * @param value the value to divide by
+ * * @return the stack after the division
+ * **/
+Stack& Stack::operator/=(int value)
+{
+	for (auto& element : m_stack) {
+		element /= value;
+	}
+	return *this;
+}
+
+/**
+ * operator + concatenate two stacks
+ * @param other the stack to concatenate
+ * @return the new stack after the concatenation
+ **/
+Stack Stack::operator+(const Stack& other) const
+{
+	//concatinate the two stacks
+	Stack newStack(*this);
+	for (auto& element : other.m_stack) {
+		newStack.m_stack.push_back(element);
```

Comment:
Instead of using a private member of `newStack`, you just use the `push` method you implemented

Reply:
fixed